### PR TITLE
[DSVA-2528] Implement othe IoU losses

### PR DESCRIPTION
**Pull request by pkolebski**

### CODE REVIEW

**Comment by pkolebski** [comment]
In `N:\ComputerVision\projects\datasets_cleaning\datasets\05_cumultative_training\models\test_subdataset5_ciou/train_log.txt` you can find traceback

File: yolox/models/losses.py [hunk 1]
```diff
@@ -1,49 +1,68 @@
...
+        (box1[:, :2] + box1[:, 2:] / 2), (box2[:, :2] + box2[:, 2:] / 2)
+    )
+    return box_top_left, box_bottom_right
+
+
+def get_d_c_parameters(box1, box2) -> tuple[torch.Tensor, torch.Tensor]:
+    d = (box1[:, :2] - box2[:, :2]).pow(2).sum(1).sqrt()
+    c_tl, c_br = get_smalest_enclosing_box(box1, box2)
+    c = (c_br - c_tl).pow(2).sum(1).sqrt()
+    return d, c
```
Comment (Pawel842) [inline yolox/models/losses.py:23]: imo to make sure it's proper calculation that methods needs unit tests.

**Review by Pawel842 — CHANGES_REQUESTED** [review]

**Review by AnhelinaRusak — COMMENTED** [review]
Just add tests, as Paweł mentioned

File: yolox/data/datasets/coco.py [hunk 2]
```diff
@@ -157,6 +158,9 @@ def load_image(self, index):
         image_file_path = self.image_paths[image_file_name]
 
         image = cv2.imread(image_file_path)
+        if image is None:
+            time.sleep(3)
+            image = cv2.imread(image_file_path)
         assert image is not None, f"file named {image_file_path} not found"
```
Comment (pkolebski) [inline yolox/data/datasets/coco.py:164]: I had often problems with loading images - probably issues with VPN/network drive connection. With this "fix" I could run full training without error.

**Review by Pawel842 — APPROVED** [review]

**Review by AnhelinaRusak — APPROVED** [review]

**Pull request merged**